chatgpt/api.py

```diff
@@ -141,6 +156,40 @@ def new_conversation(self) -> None:
         self._conversation_id = None
         self._parent_message_id = _generate_uuid()
 
+    @staticmethod
+    def __get_class_logger() -> logging.Logger:
+        class __IOFormatter(logging.Formatter):
+            """Used to specify custom logging keys."""
+
+            def format(self, record):
+                record.timestamp = record.args.get("timestamp")
+                record.input = record.args.get("input")
+                record.output = record.args.get("output")
+                record.id = record.args.get("id")
+                record.conversation_id = record.args.get("conversation_id")
+                record.parent_message_id = record.args.get("parent_message_id")
+                return super().format(record)
+
+        logger = logging.getLogger("ChatGPT")
+        io_json_formatter = __IOFormatter(
+            '{"timestamp":"%(timestamp)s",\
+            "input": "%(input)s", "output": "%(output)s",\
+            "id": "%(id)s", "conversation_id": "%(conversation_id)s",\
+            "parent_message_id": "%(parent_message_id)s"}'
+        )
+        time_tuple = time.localtime(time.time())
+        time_string = time.strftime("%H:%M:%S", time_tuple)
+        if not LOGGING_DIR.exists():
+            LOGGING_DIR.mkdir(parents=True, exist_ok=True)
+        logging_path = LOGGING_DIR / f"chatgpt_logs_{time_string}.log"
+        file_handler = logging.FileHandler(
+            filename=f"{logging_path}", mode="w"
```

Comment:
`f"{logging_path}"` - is it about converting it to a string?
If so, I think that converting it explicitly would be a better approach, having in mind Zen of Python rule (explicit is better than implicit) - `str(logging_path)`

Reply:
logging_path is now explicitly converted to str.